src/Pos/Block/Pure.hs

```diff
@@ -584,4 +625,4 @@ verifyBlocks curSlotId verifyNoUnknown bvd initLeaders = view _3 . foldl' step s
                 , vbpMaxSize = Just (bvdMaxBlockSize bvd)
                 , vbpVerifyNoUnknown = verifyNoUnknown
                 }
-        in (newLeaders, Just $ getBlockHeader blk, res <> verifyBlock vbp blk)
+        in (newLeaders, newPsks, Just $ getBlockHeader blk, res <> verifyBlock vbp blk)
```

Comment:
Ah, yes, this needs some clarification. So `GState` holds the complete map `i -> d`, but when verifying a certain set of blocks, we only care about certificates of block issuers. Because that's what check will boil down to eventually. So instead of getting the whole set of psks from `GState` (might be big), I retrieve a subset.